File: phpRatchetWebSocket/Services/UserInfoFetchingService.php
<?php

namespace Services;

use Illuminate\Support\Facades\Http;


class UserInfoFetchingService {
    

    public function __construct() {}


    public function getUsernameOfUser(int $userId) {
        try {
            $response = Http::withHeaders([
                'Content-Type' => 'application/json',
            ])->post(
                'http://34.111.89.101/api/Home-Page/laravelBackend1/graphql',
                [
                    'query' => 'query ($userId: Int!) {
                        getUsernameOfUserIdFromWebSocket(userId: $userId)
                    }',
                    'variables' => [
                        'userId' => $userId
                    ]
                ]
            );

            if ($response->failed()) {
                return "user $userId";
            }

            $stringifiedResponseData = $response->body();
            $parsedResponseData = json_decode($stringifiedResponseData, true);

            $usernameOfUser = $parsedResponseData['data']['getUsernameOfUserIdFromWebSocket'];
            return $usernameOfUser;
        }
        catch (\Exception) {
            return "user $userId";
        }
    }
}

File: pythonWebSocket/services/UserInfoFetchingService.py
import requests


def get_username_of_user(user_id):
    try:
        url = 'http://34.111.89.101/api/Home-Page/laravelBackend1/graphql'
        headers = {
            'Content-Type': 'application/json',
        }
        payload = {
            'query': '''
                query ($userId: Int!) {
                    getUsernameOfUserIdFromWebSocket(userId: $userId)
                }
            ''',
            'variables': {
                'userId': user_id
            }
        }

        response = requests.post(url, json=payload, headers=headers)

        if not response.ok:
            return f"user {user_id}"

        parsed_response_data = response.json()
        username_of_user = parsed_response_data['data']['getUsernameOfUserIdFromWebSocket']
        return username_of_user

    except Exception:
        return f"user {user_id}"
